lib/vyasa_web/live/source_live/chapter/index.ex

```diff
@@ -48,28 +47,30 @@ defmodule VyasaWeb.SourceLive.Chapter.Index do
   defp sync_session(socket), do: socket
 
   defp apply_action(socket, :index, %{"source_title" => source_title, "chap_no" => chap_no} = _params) do
-    chap  = %{verses: verses, translations: [ts | _]} = Written.get_chapter(chap_no, source_title, @default_lang)
-
-    socket
-    |> stream(:verses, verses)
-    |> assign(:source_title, source_title)
-    |> assign(:chap, chap)
-    |> assign(:selected_transl, ts)
-    |> assign_meta()
+    with %Source{id: sid} = source <- Written.get_source_by_title(source_title),
+         %{verses: verses, translations: [ts | _]} = chap  <- Written.get_chapter(chap_no, sid, @default_lang) do
+
+      socket
+      |> stream(:verses, verses)
+      |> assign(:src, source)
+      |> assign(:chap, chap)
+      |> assign(:selected_transl, ts)
+      |> assign_meta()
+    else
+      _ -> raise VyasaWeb.ErrorHTML.FourOFour, message: "Chapter not Found"
```

Comment:
(3) QQ: seems like this is the general pattern for how to explicitly throw error cases to error boundaries when handling non-happy paths within a live_view component?

Reply:
raise the error and the error handler handles it. 

this is handled via the fallback_controller.ex, the actual handling defined by the message that is being passed
![image](https://github.com/ve1ld/vyasa/assets/38996397/16da0d29-1108-4ec7-8faf-e678b9a7143b)


the actual error boundary defined @ config.exs where we indicate how the error gets rendered for html/json 

```elixir
# Configures the endpoint
config :vyasa, VyasaWeb.Endpoint,
  url: [host: "localhost"],
  adapter: Phoenix.Endpoint.Cowboy2Adapter,
  render_errors: [
    formats: [html: VyasaWeb.ErrorHTML, json: VyasaWeb.ErrorJSON],
    layout: false
  ],
  pubsub_server: Vyasa.PubSub,
  live_view: [signing_salt: "61KScf/3"]

#
```


yea, the error boundary can be tested locally by changing the following  

   http: [ip: {0, 0, 0, 0}, port: System.get_env("PORT") || 4000],
   check_origin: false,
   code_reloader: true,
-  debug_errors: true,
+  debug_errors: false,
   secret_key_base: "TSFFKfQoGKCAjI5yxwt9DmxwnJg1U7CnuCsxpWjr+DqXs8EavSxtO3l3wk2PquSW",
   watchers: [
     esbuild: {Esbuild, :install_and_run, [:default, ~w(--sourcemap=inline --watch --loader:.ttf=file)]},
modified   lib/vyasa_web/controllers/og_image_controller.ex